### PR TITLE
helm processing info

### PR DESCRIPTION
**Pull request by jeffreygolden**
Adding `peek inside the box` info for how Helm charts are processed using Native Helm

### CODE REVIEW

File: content/vendor/helm/helm-processing.md
Comment (JustinN-Replicated) [inline content/vendor/helm/helm-processing.md:22]: Helm v2 has already been removed from helm support: https://helm.sh/blog/helm-v2-deprecation-timeline/
I think we can just leave the line as it was.

**Comment by JustinN-Replicated** [comment]
Right now https://kots.io/vendor/helm/helm-processing/ was exclusive to the existing helm workflow; would it make sense to have this content in its own under native helm (instead of creating another doc with shared details)?

File: content/vendor/helm/helm-processing.md
Comment (morningvera) [inline content/vendor/helm/helm-processing.md:58]: `(Picture)` ?

**Review by morningvera — APPROVED** [review]
A few minor comments but otherwise the content flows nicely and is easy to understand imo

File: content/vendor/helm/helm-processing.md
Comment (jeffreygolden) [reply content/vendor/helm/helm-processing.md:8]: I felt like without the line breaks, the paragraphs were too thick and started to become a wall of text.  Including an empty line between the sentences disrupted the flow, so this seemed like a good happy medium.  For your plugin, is it possible to exclude .md files?  The preferred method for new lines is double-whitespace https://www.markdownguide.org/basic-syntax/#line-breaks

Comment (morningvera) [reply content/vendor/helm/helm-processing.md:8]: with text wrapping, i've never felt like it reads poorly and consecutive lines without double line breaks render to single paragraphs, right? it's just personal preference though so i suggest leaving as is

Comment (jeffreygolden) [reply content/vendor/helm/helm-processing.md:8]: Double whitespace at the end of the line means "new line in this paragraph".  So these sentences are separated by a single line break on the page, like this:
image

Helps break the wall of text feel of the paragraphs, I think

Comment (samantha-replicated) [reply content/vendor/helm/helm-processing.md:8]: I use Atom as my preferred editor and it automatically deletes trailing whitespaces when you commit a file.

Comment (samantha-replicated) [reply content/vendor/helm/helm-processing.md:8]: For the line breaks, if we're trying to split the paragraph for clarity and to give the reader space ....the general default is to create 2 paragraphs vs moving a new sentence to the next line.

Comment (jeffreygolden) [reply content/vendor/helm/helm-processing.md:8]: I can replace the double spaces with <br> as suggested by the markdown guide (https://www.markdownguide.org/basic-syntax/#line-breaks) if that's preferred.  Though if markdown officially supports trailing whitespace as a formatting function, you could look into excluding .md files from the auto-formatting in atom, like: 
image

Comment (jeffreygolden) [reply content/vendor/helm/helm-processing.md]: I've restructured the PR to eliminate all single line breaks.

File: content/vendor/helm/helm-processing.md
Comment (jeffreygolden) [reply content/vendor/helm/helm-processing.md]: placeholder, removed :)

File: content/vendor/helm/helm-processing.md
Comment (JustinN-Replicated) [reply content/vendor/helm/helm-processing.md]: The new naming is somewhat TBD so let's keep what we're doing now and adjust it all at once I think. Right now, the leading candidates are: 
"_Installing via Helm_" (to replace native)
"_Installing via kubectl apply_" (to replace replicated kots helm)

Comment (JustinN-Replicated) [reply content/vendor/helm/helm-processing.md]: Would it make sense if we explain _why_ it doesn't rely on tiller? The real reason is because replicated is creating the resources not helm (via apply) afaik, right?

Comment (JustinN-Replicated) [reply content/vendor/helm/helm-processing.md]: @jonquil2002  I haven't bounced these off of docs team yet - my apologies for that. These were the product of a few discussions with leadership; the goal is to 'soften' our language a bit with helm as the current names may imply too much detail.

**Review by morningvera — APPROVED** [review]
lgtm

File: content/vendor/helm/helm-processing.md
Comment (samantha-replicated) [inline content/vendor/helm/helm-processing.md:39]:
```suggestion
Replicated's [`HelmChart` spec](https://kots.io/reference/v1beta1/helmchart/) can be modified to allow the `ConfigOptions` to [overwrite a chart's values](https://kots.io/reference/v1beta1/helmchart/#values). This allows vendors to surface a chart's value options inside the Replicated Config page. After Replicated templating is processed on the `values.yaml` file, all files from the original Helm tarball are written to the `base/charts/` directory, maintaining the original directory structure of the Helm Chart. A `kustomization.yaml` file is included in each chart and sub chart directory. This is used later to merge kustomization instructions up to the chart resources.
```

File: content/vendor/helm/helm-processing.md
Comment (samantha-replicated) [inline content/vendor/helm/helm-processing.md:41]:
```suggestion
**Example**
In this example, a `base` directory is illustrated and the `base/charts/postgresql/kustomization.yaml` includes the default postgresql chart:
```

File: content/vendor/helm/helm-processing.md
Comment (samantha-replicated) [inline content/vendor/helm/helm-processing.md:62]:
```suggestion
**Example**
In this example, a `midstream` directory is illustrated and the `overlays/midstream/charts/postgresql/kustomization.yaml` includes the default postgresql chart:
```

File: content/vendor/helm/helm-processing.md
Comment (samantha-replicated) [inline content/vendor/helm/helm-processing.md:91]:
```suggestion
**Example**
In this example, a `downstream` directory is illustrated and the `overlays/downstream/this-cluster/charts/postgresql/kustomization.yaml` includes the default postgresql chart:
```

**Review by samantha-replicated — CHANGES_REQUESTED** [review]
Hi Jeff, 

Great content. I had a few little changes and organizational suggestions. I didn't commit anything so you could ensure my changes didn't miss the mark for technical accuracy. Let me know when you're done and I can fix the approval so we can get this card merged :)

File: content/vendor/helm/helm-processing.md
Comment (jonquil2002) [reply content/vendor/helm/helm-processing.md]: @JustinN-Replicated  No worries, But it looks like Sammi has picked this up for review, so I'll leave it to her.

File: content/vendor/helm/helm-processing.md
Comment (jeffreygolden) [reply content/vendor/helm/helm-processing.md:19]: This was already on the page before I added my content.  I agree it's probably not worth stating here, I think we've worked past this at a technical level by automatically detecting the chart version anyway.

Since it has already been on the page for some time, are we OK to remove this, or does that become another discussion?

File: content/vendor/helm/helm-processing.md
Comment (jeffreygolden) [reply content/vendor/helm/helm-processing.md:44]:
```suggestion
**Example**
	In this example, a `base` directory is illustrated and the `base/charts/postgresql/kustomization.yaml` targets chart resources:
```

File: content/vendor/helm/helm-processing.md
Comment (jeffreygolden) [reply content/vendor/helm/helm-processing.md:65]:
```suggestion
**Example**
	In this example, a `midstream` directory is illustrated and the `overlays/midstream/charts/postgresql/kustomization.yaml` adds Replicated customization:
```

File: content/vendor/helm/helm-processing.md
Comment (jeffreygolden) [reply content/vendor/helm/helm-processing.md:94]:
```suggestion
**Example**
	In this example, a `downstream` directory is illustrated and the `overlays/downstream/this-cluster/charts/postgresql/kustomization.yaml` provides a template for last-mile overrides:
```

**Review by jeffreygolden — COMMENTED** [review]
@samantha-replicated Suggested a slight change to the example text to be more specific about what each `kustomization.yaml` file is showing.  Feel free to edit my suggestion :)

File: content/vendor/helm/helm-processing.md
Comment (JustinN-Replicated) [reply content/vendor/helm/helm-processing.md:19]: Do we explicitly state that we use v2 by default? If not, we should. This alludes to that fact also it may be unexpected that this is the case since v2 is EOL (I am presuming we have not changed because it could be a breaking change?)

File: content/vendor/helm/helm-processing.md
Comment (JustinN-Replicated) [reply content/vendor/helm/helm-processing.md]: @samantha-replicated please review the mention above; it has been a challenge to find the 'right' names but worth sharing the latest with you all else considering.

File: content/vendor/helm/helm-processing.md
Comment (jeffreygolden) [reply content/vendor/helm/helm-processing.md]: technically, if left undefined and on a version of KOTS later than 1.42 or something, we will auto-detect the Helm version and use what the chart requires.  So the real `default` is "we'll figure it out and use the right one"

Comment (JustinN-Replicated) [reply content/vendor/helm/helm-processing.md]: Maybe this is the place to mention that, though we should be specific about detecting v2 and v3 only (IE to pre-empt next version of chart when v4 comes out).
Should we do one of those note/callouts about the version where this changed in KOTS as well and allude to old behavior too?

Comment (samantha-replicated) [reply content/vendor/helm/helm-processing.md]: I reviewed another PR recently from Todd where he was updating some Helm pages and Vera had called out that V2 was used by default and we ensured that was stated in that doc. I'd have to go search, but we did state that somewhere in the helm docs.

**Comment by samantha-replicated** [comment]
> 

I'm good with the changes you made and just committed :)

File: content/vendor/helm/helm-processing.md
Comment (samantha-replicated) [reply content/vendor/helm/helm-processing.md]: I agree with updating the naming conventions in one swoop vs now. I like `Installing with Helm`, I think it's encompassing enough to cover all of our helm options without being too descriptive in the title. And as a top-level heading it gives you more flexibility for nesting various related content beneath it. It also helps bucket the content into a category via the TOC, so users can scan it and dig into the Helm content vs questioning what "type" of helm installation then need to read first. 

The 2nd option is a little sticky for me. Speaking purely from a UX perspective in the docs. Would a user read that title and know what that means? If we write an overview topic to introduce it first, then It should be ok. Something like, "We have 2 different helm installs.....this does.....that does..." Then a task topic for each install would nest nicely there. Without that, the title doesn't standalone as clearly. With no reference to Helm in the title, someone scanning the TOC looking for that info would likely pass it on first glance. We also run into SEO ramifications possibly if they are googling how to install with Helm and not hitting this topic as a result.

**Comment by samantha-replicated** [comment]
I'm gonna go ahead and mark my approval, because I have to run for daycare PU and don't want to hold up merging. The last two items have lots of convo around them and aren't an impediment to publishing the content.

**Review by samantha-replicated — APPROVED** [review]

**Pull request merged**
